### PR TITLE
Setting use_beeline by default for hive cli connection

### PR DESCRIPTION
**Pull request by amoghrajesh**
Hive connections from airflow are usually always made through beeline in most cases, due to this we should enable the beeline checkbox by default to make it more convenient to not miss the chance to tick this or while deploying the connections through CLI or API.

The code also follows the pattern of "if beeline, do this"
```
    def _prepare_cli_cmd(self) -> list[Any]:
        """Create the command list from available information."""
        conn = self.conn
        hive_bin = "hive"
        cmd_extra = []

        if self.use_beeline:
            hive_bin = "beeline"
            self._validate_beeline_parameters(conn)
            if self.high_availability:
                jdbc_url = f"jdbc:hive2://{conn.host}/{conn.schema}"
                self.log.info("High Availability set, setting JDBC url as %s", jdbc_url)
            else:
                jdbc_url = f"jdbc:hive2://{conn.host}:{conn.port}/{conn.schema}"
                self.log.info("High Availability not set, setting JDBC url as %s", jdbc_url)
            if conf.get("core", "security") == "kerberos":
                template = conn.extra_dejson.get("principal", "hive/_HOST@EXAMPLE.COM")
                if "_HOST" in template:
                    template = utils.replace_hostname_pattern(utils.get_components(template))
                proxy_user = self._get_proxy_user()
                if ";" in template:
                    raise RuntimeError("The principal should not contain the ';' character")
                if ";" in proxy_user:
                    raise RuntimeError("The proxy_user should not contain the ';' character")
                jdbc_url += f";principal={template};{proxy_user}"
                if self.high_availability:
                    if not jdbc_url.endswith(";"):
                        jdbc_url += ";"
                    jdbc_url += "serviceDiscoveryMode=zooKeeper;ssl=true;zooKeeperNamespace=hiveserver2"
            elif self.auth:
                jdbc_url += ";auth=" + self.auth

            jdbc_url = f'"{jdbc_url}"'

            cmd_extra += ["-u", jdbc_url]
            if conn.login:
                cmd_extra += ["-n", conn.login]
            if conn.password:
                cmd_extra += ["-p", conn.password]

        hive_params_list = self.hive_cli_params.split()

        return [hive_bin, *cmd_extra, *hive_params_list]
```

Also added an additional unit test for default values

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/apache/hive/hooks/hive.py [hunk 1]
```diff
@@ -126,7 +126,7 @@ def get_connection_form_widgets(cls) -> dict[str, Any]:
         from wtforms import BooleanField, StringField
 
         return {
-            "use_beeline": BooleanField(lazy_gettext("Use Beeline"), default=False),
+            "use_beeline": BooleanField(lazy_gettext("Use Beeline"), default=True),
```
Comment (eladkal) [inline airflow/providers/apache/hive/hooks/hive.py:129]: This requires major release as it consider a breaking change isn't it?

Comment (amoghrajesh) [reply airflow/providers/apache/hive/hooks/hive.py:129]: Why is it viewed as a breaking change? Would change in default value be that major?

Comment (amoghrajesh) [reply airflow/providers/apache/hive/hooks/hive.py:129]: Got the information from slack

File: airflow/providers/apache/hive/CHANGELOG.rst [hunk 2]
```diff
@@ -27,6 +27,14 @@
 Changelog
 ---------
 
+7.0.2
```
Comment (potiuk) [inline airflow/providers/apache/hive/CHANGELOG.rst]: I think it's a breaking change it should be 8.0.0

Comment (amoghrajesh) [reply airflow/providers/apache/hive/CHANGELOG.rst]: Sure, made the change for that

File: airflow/providers/apache/hive/provider.yaml [hunk 3]
```diff
@@ -25,6 +25,7 @@ state: ready
 source-date-epoch: 1709554960
 # note that those versions are maintained by release manager - do not update them manually
 versions:
+  - 7.0.2
```
Comment (potiuk) [inline airflow/providers/apache/hive/provider.yaml]:
```suggestion
  - 8.0.0
```

Comment (potiuk) [reply airflow/providers/apache/hive/provider.yaml]: And spelling issue.

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Approved with NIT's

File: airflow/providers/apache/hive/CHANGELOG.rst [hunk 4]
```diff
@@ -27,6 +27,19 @@
...
+.....
+
+
+Breaking changes
+~~~~~~~~~~~~~~~~
+
+Changed the default value of ``use_beeline`` in hive cli connection to True.
+Beeline will be always enabled by default in this connection type from now onwards.
+
+* ``Setting use_beeline by default for hive cli connection (#38763)``
```
Comment (eladkal) [inline airflow/providers/apache/hive/CHANGELOG.rst]: This will be set by  generating docs automaticly
```suggestion
```

**Review by eladkal — APPROVED** [review]

File: airflow/providers/apache/hive/CHANGELOG.rst [hunk 5]
```diff
@@ -27,6 +27,18 @@
...
 
+8.0.0
+.....
+
+
+Breaking changes
+~~~~~~~~~~~~~~~~
+
+Changed the default value of ``use_beeline`` in hive cli connection to True.
+Beeline will be always enabled by default in this connection type from now onwards.
```
Comment (eladkal) [inline airflow/providers/apache/hive/CHANGELOG.rst]:
```suggestion
Beeline will be always enabled by default in this connection type
```

Fix spelling check

**Comment by amoghrajesh** [comment]
@eladkal fixed spelling and spaces, we should get a green run now.

**Review by eladkal — APPROVED** [review]

**Pull request merged**